indra/sources/signor.py

```diff
@@ -0,0 +1,379 @@
+                                data={'organism':'human', 'format':'csv',
+                                      'submit':'Download'})
+            if res.status_code == 200:
+                csv_reader = csv.reader(StringIO(res.text), delimiter=delimiter,
+                                        quoting=csv.QUOTE_MINIMAL)
+                next(csv_reader) # Skip the header row
+                self._data = [SignorRow(*[f.strip() for f in r])
+                              for r in csv_reader]
+            else:
+                raise Exception('Could not download Signor data.')
+        # Process into statements
+        self.statements = []
+        self.no_mech_rows = []
+        for row in self._data:
+            row_stmts, no_mech = self._process_row(row)
+            if no_mech:
+                self.no_mech_rows.append(row)
+            self.statements.extend(row_stmts)
+        # No-mechanism rows by mechanism type
+        no_mech_ctr = Counter([row.MECHANISM for row in self.no_mech_rows])
+        self.no_mech_ctr = sorted([(k, v) for k, v in no_mech_ctr.items()],
+                                  key=lambda x: x[1], reverse=True)
+
+    @staticmethod
+    def _get_agent(ent_name, ent_type, id, database):
+        gnd_type = _type_db_map[(ent_type, database)]
+        if gnd_type == 'UP':
+            up_id = id
+            db_refs = {'UP': up_id}
+            name = uniprot_client.get_gene_name(up_id)
+            hgnc_id = hgnc_client.get_hgnc_id(name)
+            if hgnc_id:
+                db_refs['HGNC'] = hgnc_id
+        # Other possible groundings are PUBCHEM and SIGNOR
+        elif gnd_type is not None:
+            assert database in ('PUBCHEM', 'SIGNOR')
+            db_refs = {database: id}
+            name = ent_name
+        # If no grounding, include as an untyped/ungrounded node
+        else:
+            name = ent_name
+            db_refs = {}
+        return Agent(name, db_refs=db_refs)
+
+    @staticmethod
+    def _get_evidence(row):
+        # Get epistemics (direct/indirect)
+        epistemics = {}
+        epistemics['direct'] = True if row.DIRECT == 'YES' else False
+        # Get annotations
+        _n = lambda s: s if s else None
+        # TODO: Refactor to exclude keys that are just Nones
+        annotations = {
+                'SEQUENCE': _n(row.SEQUENCE),
+                'TAX_ID': _n(row.TAX_ID),
+                'CELL_DATA': _n(row.CELL_DATA),
+                'TISSUE_DATA': _n(row.TISSUE_DATA),
+                'MODULATOR_COMPLEX': _n(row.MODULATOR_COMPLEX),
+                'TARGET_COMPLEX': _n(row.TARGET_COMPLEX),
+                'MODIFICATIONA': _n(row.MODIFICATIONA),
+                'MODASEQ': _n(row.MODASEQ),
+                'MODIFICATIONB': _n(row.MODIFICATIONB),
+                'MODBSEQ': _n(row.MODBSEQ),
+                'NOTES': _n(row.NOTES),
+                'ANNOTATOR': _n(row.ANNOTATOR)}
+        return Evidence(source_api='SIGNOR', source_id=row.SIGNOR_ID,
```

Comment:
Not a big deal but we use lowercase for all other sources, so 'signor' might be better. Could you add some parameters to `indra.belief.__init__.py` for 'signor' too?